components/create_comment/create_comment.tsx

```diff
@@ -1247,6 +1255,8 @@ class CreateComment extends React.PureComponent<Props, State> {
                                 badConnection={this.props.badConnection}
                                 listenForMentionKeyClick={true}
                                 useChannelMentions={this.props.useChannelMentions}
+                                customEditor={this.state.customEditor}
+                                setCustomEditor={this.setCustomEditor}
```

Comment:
In order to de-duplicate code, I suggest that we instead use a `supportCustomEditors` boolean here instrad of adding the two props in the highlighted code. Then we can have the child component handle all of the logic of loading the components from redux, and manage the currently selected editor.

Reply:
I'd almost wonder if we could just move the Help and Preview text into the Textbox component to avoid duplicating it across all 3 of these places. I'm not entirely sure why it's outside of that to begin with. That way, CreatePost and friends wouldn't even need to know about the custom editor

Reply:
> I'm not entirely sure why it's outside of that to begin with

@hmhealey I assume there are some places where we use `Textbox` without wanting to show the `TextboxLinks` component? I know we export `Textbox` to plugins to use as they see fit. Maybe we don't want to show them in that instance. We can also provide a `showTextboxLinks` prop to the component to differentiate.

The placement of the links may not be appropriate to have scoped in the same DOM element region as the `textarea`, as there are elements in between these two components, such as "User is typing".

Reply:
Oh, I thought the "User is Typing" message was part of TextboxLinks as well since I just assumed it was everything under the post textbox. Ideally, those would probably all be part of the same component, but I guess this is still something we have to deal with. I think @larkox's work on https://github.com/mattermost/mattermost-webapp/pull/9944 will help with that, so we could leave some of the duplication here in the mean time until that's in and then sort it out.

I think we'd probably want those links to be part of that component in the long term since I think it's the core of the post texbox, but you're right that it might be used differently elsewhere. I think the reason that it's exported is because Playbooks uses it to autocomplete slash commands or something similar. I think that actually falls outside this component's intended use case, but we're stuck with that until we get some better common components.

Anyway, I guess we can leave it as-is and figure out de-duping it later